openedx_events/tooling.py

```diff
@@ -185,6 +157,86 @@ def validate_sender():
 
         return responses
 
+    def send_event(self, send_robust=True, time=None, **kwargs):
+        """
+        Send events to all connected receivers.
+
+        Arguments:
+            send_robust (bool): Defaults to True. See Django signal docs.
+            time (datetime): (Optional - see note) Timestamp when the event was
+                sent with UTC timezone. For events requiring a DB create or
+                update, use the timestamp from the DB record. Defaults to
+                current time in UTC. This argument is optional for backward
+                compatability, but ideally would be explicitly set. See OEP-41
+                for details.
+
+        Used to send events just like Django signals are sent. In addition,
+        some validations are executed on the arguments, and then generates relevant
+        metadata that can be used for logging or debugging purposes. Besides this behavior,
+        send_event behaves just like the send method.
+
+        If the event is disabled (i.e _allow_events is False), then this method
+        won't have any effect. Meaning, the Django Signal won't be sent.
+
+        Example usage:
+            >>> STUDENT_REGISTRATION_COMPLETED.send_event(
+                user=user_data, registration=registration_data,
+            )
+            [(<function callback at 0x7f2ce638ef70>, 'callback response')]
+
+        Keyword arguments:
+            send_robust (bool): determines whether the Django signal will be
+            sent using the method `send` or `send_robust`.
+
+        Returns:
+            list: response of each receiver following the format
+            [(receiver, response), ... ]. Empty list if the event is disabled.
+
+        Exceptions raised:
+            SenderValidationError: raised when there's a mismatch between
+            arguments passed to this method and arguments used to initialize
+            the event.
+        """
+        metadata = self.generate_signal_metadata(time=time)
+        return self._send_event_with_metadata(metadata=metadata, send_robust=send_robust, **kwargs)
+
+    def send_event_with_provided_metadata(
```

Comment:
[nit] I prefer `custom` because when I see `provided` it makes me think that the data is somehow provided externally (as opposed to being set by the caller). Preference is not enough to block the review, however.